main/src/library/Foldable.flix

```diff
@@ -219,23 +219,26 @@ pub class Foldable[t : Type -> Type] {
     ///
     /// Returns an iterator over `t`.
     ///
-    pub def iterator(r: Region[r], t: t[a]): Iterator[a, r] \ Write(r) =
+    pub def iterator(r: Region[r], t: t[a]): Iterator[a, r, r] \ Write(r) =
         let arr = Foldable.toArray(r, t);
```

Comment:
Can this be avoided? Or is it not yet the time?

Reply:
Updated - removed the use of an intermediate array.